bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -312,12 +312,13 @@ static ICoreWebView2CookieManager getCookieManager() {
 		SWT.error(SWT.ERROR_NOT_IMPLEMENTED, null, " [WebView2: cookie access requires a Browser instance]");
 	}
 	Edge instance = environmentWrapper.instances().get(0);
-	if (instance.webView_2 == null) {
+	if (instance.lastWebView_2Task.isCancelled()) {
 		SWT.error(SWT.ERROR_NOT_IMPLEMENTED, null, " [WebView2 version 88+ is required to access cookies]");
 	}
 
 	long[] ppv = new long[1];
-	int hr = instance.webView_2.get_CookieManager(ppv);
+	instance.waitForWebviewInitialization();
+	int hr = instance.lastWebView_2Task.join().get_CookieManager(ppv);
```

Comment:
Using these `last*Task.join()` and `last*Task.isCancelled()` calls all of the class reduces comprehensibility. This should be wrapped into some more self-explaining method/object, such that you call some `getInitializedWebView()` and `wasInitializationCancelled()` (or comparable) and wrap the way in which joining and cancelling futures is done into those methods.